internal/service/cloudbackupschedule/resource_cloud_backup_schedule.go

```diff
@@ -674,3 +740,79 @@ func policyItemID(policyState map[string]any) *string {
 	}
 	return nil
 }
+
+func shouldUseOldAPI(d *schema.ResourceData, isCreate bool) (bool, error) {
+	copySettings := d.Get("copy_settings")
+	if isCopySettingsNonEmptyOrChanged(d) {
+		return CheckCopySettingsToUseOldAPI(copySettings.([]any), isCreate)
+	}
+	return false, nil
+}
+
+func isCopySettingsNonEmptyOrChanged(d *schema.ResourceData) bool {
+	copySettings := d.Get("copy_settings")
+	return copySettings != nil && (conversion.HasElementsSliceOrMap(copySettings) || d.HasChange("copy_settings"))
+}
+
+// CheckCopySettingsToUseOldAPI verifies that all elements in tfList use either `replication_spec_id` or `zone_id`
+// Returns an error if any element has both `replication_spec_id` and `zone_id` set during create
+// and returns a bool if the old API should be used or not
+func CheckCopySettingsToUseOldAPI(tfList []any, isCreate bool) (bool, error) {
+	allHaveRepID := true
+
+	for _, tfMapRaw := range tfList {
+		tfMap, ok := tfMapRaw.(map[string]any)
+		if !ok {
+			return false, fmt.Errorf("element is not a valid map[string]any")
+		}
+
+		repSpecID, repOk := tfMap["replication_spec_id"].(string)
+		zoneID, zoneOk := tfMap["zone_id"].(string)
+
+		if repOk && repSpecID != "" && zoneOk && zoneID != "" {
+			if isCreate {
+				return false, fmt.Errorf("both 'replication_spec_id' and 'zone_id' cannot be set unless resource is being updated")
```

Comment:
Can we use ConflictsWith validation in the schema? To simplify this logic

Reply:
I added that initially but had to remove that and add manual validation instead. 
The issue is that when user is migrating from using rep_spec_id to zone_id then both are present initially when update is called (even if users remove rep_spec_id from their config & add zone_id, since rep_spec_id is O/C, it stays in the schema with the value from the state). It is only after the update has completed that rep_spec_id is removed from the object.

To support the above scenario, we can't use ConflictsWith because it would throw an error when users try to update their configs to the new schema.

Reply:
I see, was suspecting that but was not sure if ConflictsWith considered values present only in the state.